tests/common/include/tests/common/path_correction.inl

```diff
@@ -0,0 +1,524 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "detray/definitions/units.hpp"
+#include "detray/materials/predefined_materials.hpp"
+#include "detray/propagator/actor_chain.hpp"
+#include "detray/propagator/actors/parameter_resetter.hpp"
+#include "detray/propagator/actors/parameter_transporter.hpp"
+#include "detray/propagator/line_stepper.hpp"
+#include "detray/propagator/navigator.hpp"
+#include "detray/propagator/propagator.hpp"
+#include "detray/propagator/rk_stepper.hpp"
+#include "detray/tracks/tracks.hpp"
+#include "tests/common/tools/detector_metadata.hpp"
+
+// Covfie include(s).
+#include <covfie/core/field.hpp>
+#include <covfie/core/field_view.hpp>
+
+// Vecmem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// google-test include(s).
+#include <gtest/gtest.h>
+
+using namespace detray;
+
+struct surface_targeter : actor {
+
+    struct state {
+        scalar _path;
+        dindex _target_surface_index = dindex_invalid;
+    };
+
+    /// Enforces thepath limit on a stepper state
+    ///
+    /// @param abrt_state contains the path limit
+    /// @param prop_state state of the propagation
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE void operator()(state &actor_state,
+                                       propagator_state_t &prop_state) const {
+
+        prop_state._heartbeat = true;
+        auto &navigation = prop_state._navigation;
+        auto &stepping = prop_state._stepping;
+        navigation.set_full_trust();
```

Comment:
This should not be possible, no actor should be able to raise the trust level arbitrarily

Reply:
My meaning is, this does not force the trust level to ```e_full_trust```, if the stepper reduced the trust level first. It might not have the effect you seem to expect here

```
/// Update navigation trust level to full trust
DETRAY_HOST_DEVICE
inline void set_full_trust() {
    _trust_level = _trust_level <= navigation::trust_level::e_full
                       ? _trust_level
                       : navigation::trust_level::e_full;
}
```

This behaviour is very much intended.

Reply:
Come to think of it, this method should be removed altogether. I literally has no effect, lol